src/applications/personalization/dashboard/actions/messaging.js

```diff
@@ -15,49 +15,61 @@ import {
 
 const baseUrl = `${environment.API_URL}/v0/messaging/health`;
 
-export function fetchFolder(id, query = {}) {
-  return dispatch => {
-    const errorHandler = () => dispatch({ type: FETCH_FOLDER_FAILURE });
+export const fetchFolder = (id, query = {}) => async dispatch => {
+  dispatch({
+    type: LOADING_FOLDER,
+    request: { id, query },
+  });
 
-    dispatch({
-      type: LOADING_FOLDER,
-      request: { id, query },
-    });
+  // Escape early if do not have a valid ID with which to fetch folders.
+  if (!id && id !== 0) {
+    dispatch({ type: FETCH_FOLDER_FAILURE, errors: [] });
+    return;
+  }
 
-    if (id !== null) {
-      const folderUrl = `/folders/${id}`;
-      const messagesUrl = createUrlWithQuery(`${folderUrl}/messages`, query);
+  const folderUrl = `/folders/${id}`;
+  const messagesUrl = createUrlWithQuery(`${folderUrl}/messages`, query);
 
-      if (shouldMockApiRequest()) {
-        dispatch({
-          type: FETCH_FOLDER_SUCCESS,
-          folder: mockFolderResponse,
-          messages: mockMessagesResponse,
-        });
+  if (shouldMockApiRequest()) {
+    dispatch({
+      type: FETCH_FOLDER_SUCCESS,
+      folder: mockFolderResponse,
+      messages: mockMessagesResponse,
+    });
+    return;
+  }
 
-        return;
-      }
+  try {
+    const folderResponse = await apiRequest(`${baseUrl}${folderUrl}`);
+    const messagesResponse = await apiRequest(`${baseUrl}${messagesUrl}`);
 
-      Promise.all(
```

Comment:
I removed the Promise.all for reasability's sake. The only difference in this block is the added `errorType` to both the folder and messages errors.